app/scripts/views/sign_up.js

```diff
@@ -51,6 +51,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, Session, AuthErrors,
         return p(false);
       }
 
+      /**
+       * check if there is an account we can suggest for an OAuth flow.
+       */
+      var account = this.user.getChooserAccount();
+      var serviceName = this.relier.get('serviceName');
+
+      if (account.get('email') && serviceName) {
```

Comment:
> Does this already exist and work? The commentary in #2101 seems to suggest that it's a WIP.

We have (unfortunately) two places (ref #1870) that decide which page `/` should redirect to:
* https://github.com/mozilla/fxa-content-server/blob/2e59d47082e615d556271f2f63b39ada1b169650/app/scripts/router.js#L153
* https://github.com/mozilla/fxa-content-server/blob/2e59d47082e615d556271f2f63b39ada1b169650/app/scripts/lib/app-start.js#L526

@rfk, you correctly point out that sending an OAuth user to `/` is also not really semantically correct. In addition, my assertion that `/` will do the right thing is only partially correct - users are redirected either `/settings` or `/signup`, but never to `/signin`. Sending the user to `/oauth` or `/oauth/verb-indicating-do-the-right-thing` and deciding between `/oauth/signin` and `/oauth/signup` seems like a better solution.